modules/storage/storage-accounts/blob-services/main.bicep

```diff
@@ -173,6 +176,7 @@ resource blobServices 'Microsoft.Storage/storageAccounts/blobServices@2022-09-01
     deleteRetentionPolicy: {
       enabled: deleteRetentionPolicyEnabled
       days: deleteRetentionPolicyEnabled == true ? deleteRetentionPolicyDays : null
+      allowPermanentDelete: deleteRetentionPolicyEnabled == true && deleteRetentionPolicyAllowPermanentDelete == true ? deleteRetentionPolicyAllowPermanentDelete : null
```

Comment:
When `restorePolicy` is used, `allowPermanentDelete` need to be set to `null`. If provided as `false`, the resource deployment fails with the following error.

https://github.com/MariusStorhaug/ResourceModules/actions/runs/5776225201/job/15655151602#step:4:620